File: Lab1_Knapsack/Result.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab1_Knapsack
{
    internal class Result
    {
        public List<Item> total_results;
        public double total_value
        {
            get { return total_results.Sum(item => item.value); }
        }
        public double  total_weight
        {
            get { return total_results.Sum(item => item.weight); }
        }

        public Result(List<Item> items)
        {
            total_results = items;
        }

        public override string ToString()
        {
            string result = "Items in Knapsack\n";
            foreach (var item in total_results)
            {
                result += $"Nr: {item.nr} Value: {item.value}, Weight: {item.weight}\n";
            }
            result += "\nTotal results\n";
            result += $"Total value :{total_value}, Weight: {total_weight}\n";
            return result;
        }

    }
}

File: Knapsack_java_V2/src/main/java/org/example/Result.java
package org.example;

import java.util.List;

public class Result {
    List<Integer> itemsInKnapsack;
    int totalValue;
    int totalWeight;

    public Result(List<Integer> itemsInKnapsack, int totalValue, int totalWeight) {
        this.itemsInKnapsack = itemsInKnapsack;
        this.totalValue = totalValue;
        this.totalWeight = totalWeight;
    }

    @Override
    public String toString() {
        StringBuilder result = new StringBuilder();
        result.append("Items in the knapsack:");
        for (int i : itemsInKnapsack) {
            result.append(i).append(" ");
        }
        result.append("\nTotal value: ").append(totalValue);
        result.append("\nTotal weight: ").append(totalWeight);
        return result.toString();
    }
}
